### PR TITLE
feat(comp:slider): add slider component

### LINKED ISSUE #186: [comp: slider] add slider component
- [x] I have searched the [issues](https://github.com/IduxFE/components/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?

滑动型输入器，展示当前值和可选范围。

#### Props

| 名称 | 说明 | 类型  | 默认值 | 全局配置 | 备注 |
| --- | --- | --- | --- | --- | --- |
| `v-model:value` | 绑定值 | `number \| [number, number]` | `0 \| [0, 0]` | - | - |
| `control` | 控件控制器 | `string \| number \| AbstractControl` | - | - | 配合 `@idux/cdk/forms` 使用, 参考 [Form](/components/form/zh) |
| `disabled` | 设置禁用状态 | `boolean` | `false` | - | - |
| `dots` | 显示间断点 | `boolean` | `false` | - | `marks` 间断点会始终显示 |
|`marks`|刻度标记，`key` 的类型必须为 `number` 且取值在闭区间 `[min, max]` 内，每个标签可以单独设置样式|`object`|-|-|`{ number: string \| VNode } or { number: { style: object, label: string \| VNode } } or { number: () => VNode }` |
| `keyboard` | 启用键盘行为 | `boolean` | `true` | - | - |
| `max` | 最大值 | `number` | `100` | - | - |
| `min` | 最小值 | `number` | `0` | - | - |
| `range` | 设置范围选择模式 | `boolean` | `false` | - | 双滑块模式 |
| `reverse` | 设置反向坐标轴 | `boolean` | `false` | - | - |
| `step` | 步长 | `number` | `1` | - | 要大于0 |
| `tooltipFormatter` | 格式化 `tooltip` 内容 | `(value: number) => VNode \| string \| number` | - | - | - |
| `tooltipPlacement` | 设置 `tooltip` 显示位置 | `string` | `auto` | - | 参考 [Tooltip](/components/tooltip/) |
| `tooltipVisible` | 设置 `tooltip` 显示状态 | `boolean` | - | - | `tooltip` 默认为 `hover` 和拖拽时显示，`tooltipVisible` 设置为 `true` 则始终显示，反之则始终不显示 |
| `vertical` | 设置垂直状态 | `boolean` | `false` | - | - |
| `onInput` | 拖动滑块时触发 | `(value: number) => void` | - | - | - |
| `onChange` | `Slider` 的值改变后触发(`mouseup` 阶段触发) | `(value: number) => void` | - | - | - |

### PR DESCRIPTION
**Pull request by imguolao**
fix #186

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [our guidelines](https://github.com/IDuxFE/idux/blob/main/packages/site/src/docs/Contributing.zh.md#commit)
- [ ] Tests for the changes have been added/updated or not needed
- [x] Docs and demo have been added/updated or not needed

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Component style update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Application (the showcase website) / infrastructure changes
- [ ] Other... Please describe:

### CODE REVIEW

File: packages/components/slider/style/themes/default.less [hunk 1]
```diff
@@ -0,0 +1,31 @@
+@import '../../../style/themes/default.less';
+
+@slider-marks-label-color: rgba(0, 0, 0, 0.451);
+
+@slider-rail-bg: @background-color-base;
+@slider-track-bg: #91d5ff;
```
Comment (imguolao) [inline packages/components/slider/style/themes/default.less:6]: 这个文件的颜色得换成 idux 的颜色变量。

File: packages/components/slider/style/index.less [hunk 2]
```diff
@@ -0,0 +1,187 @@
+@import './themes/default.less';
```
Comment (danranVm) [inline packages/components/slider/style/index.less:1]:
```suggestion
```
删掉，现在是在 default.less 里引入 index.less

File: packages/components/slider/style/index.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import '../../style/index.less'
```
Comment (danranVm) [inline packages/components/slider/style/index.ts:1]: 这个文件删掉, 被 themes/default.ts 替换了。

File: packages/components/slider/style/themes/default.less [hunk 4]
```diff
@@ -0,0 +1,31 @@
+@import '../../../style/themes/default.less';
+
```
Comment (danranVm) [inline packages/components/slider/style/themes/default.less:2]:
```suggestion
import '../index.less'
```

**Review by danranVm — APPROVED** [review]

**Pull request merged**